---
title: "Explore data in R"
author: "Alexey Larionov"
date: "13 Apr 2021"
output: 
  html_document:
    toc: true
    toc_float: true
    number_sections: true
editor_options: 
  chunk_output_type: console
---

# Summary

It's just a draft that hadn't been run or continued because the downstream analysis was switched to outside of AWS  

<style>
pre{
  overflow-x: auto;
}
pre code{
  word-wrap: normal;
  white-space: pre;
}
</style>

```{r echo=F}
options(width=999)
```

# Start section

```{r}

# Time
Sys.time()

# Memory
gc()

# Clean up
rm(list=ls())
graphics.off()

# Options
options(stringsAsFactors = F)

# Working folders
base_folder <- "/home/share"
data_folder <- file.path(base_folder,"data/s04_read_to_R")
scripts_folder <- file.path(base_folder,"scripts/s04_read_to_R")
setwd(scripts_folder)

# Required libraries
library(dplyr)

```

# Load data

```{r}

# VCF file name
load(file.path(data_folder,"s01_import_vcf_to_R.RData"))
base_folder <- "/home/share"
data_folder <- file.path(base_folder,"data/s04_read_to_R")
scripts_folder <- file.path(base_folder,"scripts/s04_read_to_R")

```

# Check data

```{r}

ls()

dim(vcf_header.df)
dim(variants.df)

dim(dp.mx)
dim(gq.mx)
dim(gt.mx)
dim(ad.mx)

```

# Chromosomes

```{r}

chromosomes_order <- c("chr1","chr2","chr3","chr4","chr5","chr6","chr7","chr8","chr9","chr10",
                       "chr11","chr12","chr13","chr14","chr15","chr16","chr17","chr18","chr19","chr20",
                       "chr21","chr22","chrX","chrY","chrM")
variants.df$CHROM <- ordered(variants.df$CHROM, chromosomes_order)
summary(variants.df$CHROM)
barplot(summary(variants.df$CHROM),las=2)
rm(chromosomes_order)

```

# Filter

```{r}

variants.df %>% 
  group_by(FILTER) %>% 
  summarise(counts=n()) %>% 
  arrange(desc(counts))

```

# QUAL

```{r}

quantile(variants.df$QUAL)
hist(variants.df$QUAL)
hist(variants.df$QUAL[variants.df$QUAL<250])
hist(log10(variants.df$QUAL))

```

# Final section

```{r}

sessionInfo()
Sys.time()
gc()

```
